locationsyncservice/src/main/java/com/shuttl/location_pings/config/open_lib/MockLocationHelper.kt

```diff
@@ -1,29 +1,51 @@
 package com.shuttl.location_pings.config.open_lib
 
 import android.app.Application
+import android.content.ComponentName
+import android.content.Context
 import android.content.Intent
+import android.content.ServiceConnection
+import android.os.IBinder
 import android.util.Log
 import com.shuttl.location_pings.service.MockLocationSaveService
 import com.shuttl.location_pings.config.components.LocationConfigs
+import com.shuttl.location_pings.service.LocationPingService
 
 
 object MockLocationHelper {
 
     private val TAG: String = "MockLocation"
+    var callback: LocationPingService.LocationPingServiceCallback? = null
+    private val serviceConnection by lazy {
+        object : ServiceConnection {
+            override fun onServiceDisconnected(name: ComponentName?) {
+            }
+
+            override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
+                (service as LocationPingService.CustomBinder).getService().setCallbackAndWork(callback)
+            }
+        }
+    }
 
     fun initMockLocationsModule(
         app: Application,
-        locationConfigs: LocationConfigs
+        locationConfigs: LocationConfigs,
+        callback: LocationPingService.LocationPingServiceCallback?
```

Comment:
dont think you will be needing any callbacks

Reply:
@nikhil-roy  I want both the option for mockLocation. 
- With LocationPingService : Will use initMockLocationsModule()
- Without LocationPingService : Will use startMockLocationService()

I think we should have option to start **LocationPingService** while running MockLocationService.